### PR TITLE
fix: wrong-redirection-of-pre-post-cd-details

### PR DESCRIPTION
**Pull request by rish2320**
# Description
Wrong redirection for pre/post-cd logs details

Fixes # [3160](https://dev.azure.com/DevtronLabs/Devtron/_boards/board/t/OSS%20adoption/Stories/?workitem=3160)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Tested all the cases manually.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/app/details/cicdHistory/Sidebar.tsx [hunk 1]
```diff
@@ -157,6 +159,16 @@ const HistorySummaryCard = React.memo(
             return `${generatePath(path, _params)}/${currentTab}`
         }
 
+        useEffect( () => {
+            scrollToElement()
+        }, [])
+
+        const activetriggerId = params['triggerId']
+        const scrollToElement = () => {
+            if(targetCardRef && targetCardRef.current){
```
Comment (milindk8) [inline src/components/app/details/cicdHistory/Sidebar.tsx:168]: use targetCardRef?.current instead & format the file

Comment (rish2320) [reply src/components/app/details/cicdHistory/Sidebar.tsx]: done

**Review by milindk8 — APPROVED** [review]

File: src/components/app/details/cicdHistory/Sidebar.tsx [hunk 2]
```diff
@@ -157,6 +158,16 @@ const HistorySummaryCard = React.memo(
             return `${generatePath(path, _params)}/${currentTab}`
         }
 
+        useEffect(() => {
+            scrollToElement()
+        }, [])
+
+        const activetriggerId = params['triggerId']
```
Comment (vivek-devtron) [inline src/components/app/details/cicdHistory/Sidebar.tsx:165]: rename this to activeTriggerId

File: src/components/app/details/cicdHistory/Sidebar.tsx [hunk 3]
```diff
@@ -157,6 +158,16 @@ const HistorySummaryCard = React.memo(
             return `${generatePath(path, _params)}/${currentTab}`
         }
 
+        useEffect(() => {
+            scrollToElement()
+        }, [])
```
Comment (vivek-devtron) [inline src/components/app/details/cicdHistory/Sidebar.tsx:163]: this Should have dependency of targetCardRef

File: src/components/app/details/cdDetails/CDDetails.tsx [hunk 4]
```diff
@@ -75,6 +76,12 @@ export default function CDDetails() {
             setHasMore(true)
             setHasMoreLoading(true)
         }
+        let queryString = new URLSearchParams(location.search)
+        let queryParam = queryString.get('type')
+        let deploymentStageType = queryParam === "PRECD" ? "PRE" : "POST"
```
Comment (vivek-devtron) [inline src/components/app/details/cdDetails/CDDetails.tsx:81]: Fix this condition. this will not work in case of deploy

**Review by vivek-devtron — APPROVED** [review]

File: src/components/app/details/cdDetails/CDDetails.tsx (hunk 4 above)
Comment (rish2320) [reply src/components/app/details/cdDetails/CDDetails.tsx]: done

File: src/components/app/details/cicdHistory/Sidebar.tsx (hunk 3 above)
Comment (rishabh12541) [reply src/components/app/details/cicdHistory/Sidebar.tsx]: done

File: src/components/app/details/cicdHistory/Sidebar.tsx (hunk 2 above)
Comment (rish2320) [reply src/components/app/details/cicdHistory/Sidebar.tsx]: done

**Pull request merged**